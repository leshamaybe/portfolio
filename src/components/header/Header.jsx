import React from 'react';
import Nav from '../navigation/Nav';
import Logo from '../logo/Logo';
import MarqueeBox from '../marquee/MarqueeBox';
import s from './header.module.scss';

const Header = () => {
    return (
        <header className={s.header}>
            <Logo />
            <MarqueeBox />
            <Nav />
        </header>
    );
};

export default Header;
