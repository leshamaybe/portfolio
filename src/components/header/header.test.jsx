import Header from './Header';
import { render, waitFor } from '@testing-library/react';

describe('Header', () => {
    it('Header renders correctly', async () => {
        const { container } = render(<Header />);

        await waitFor(() => {
            expect(container).toMatchSnapshot();
        });
    });
});
