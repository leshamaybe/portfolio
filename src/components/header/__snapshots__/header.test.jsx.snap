// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header renders correctly 1`] = `
<div>
  <header
    class="w-full"
  >
    <div
      class="flex flex-row justify-between items-center px-4 py-7"
    >
      <a
        class="flex items-center leading-none font-extrabold text-14+4 h-4 no-underline "
        href="/"
      >
        leshamaybe.
      </a>
      <nav
        class="max-w-[450px] w-full"
      >
        <ul
          class="flex justify-around my-4 p-0"
        >
          <li
            class="list-none"
          >
            <a
              class="bg-gradient p-[5px] text-12+3 duration-[0.25s] ease hover:bg-[length:100%_2px] hover:text-primary-black text-dark-grey"
              data-testid="About"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="list-none"
          >
            <a
              class="bg-gradient p-[5px] text-12+3 duration-[0.25s] ease hover:bg-[length:100%_2px] hover:text-primary-black text-dark-grey"
              data-testid="Resume"
              href="/resume"
            >
              Resume
            </a>
          </li>
          <li
            class="list-none"
          >
            <a
              class="bg-gradient p-[5px] text-12+3 duration-[0.25s] ease hover:bg-[length:100%_2px] hover:text-primary-black text-dark-grey"
              data-testid="Contact"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
